src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: any[], key: string): GroupsMap<Object> {
+  const obj: GroupsMap<Object> = {};
+
+  items.forEach((item) => {
+    if (!obj[item[key]]) {
+      obj[item[key]] = [item];
+    } else {
+      obj[item[key]].push(item);
+    }
+  });
+
+  return obj;
 }
```

Comment:
you should use generic here
```suggestion
export function groupByKey<T>(items: any[], key: string): GroupsMap<T> {
  const obj: GroupsMap<T> = {};

  items.forEach((item) => {
    if (!obj[item[key]]) {
      obj[item[key]] = [item];
    } else {
      obj[item[key]].push(item);
    }
  });

  return obj;
}
```